From ef56e4d882d271cdd2709c5209d7f9b54c55e8a4 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Wed, 9 Mar 2016 02:49:07 -0500
Subject: [PATCH 1/2] wr-policy: fix label for dnssec keys file

* Fix label for /etc/bind/rndc.key: named_zone_t -> dnssec_t

Remove avc denial:

  avc: denied { getattr } for pid=952 comm="rndc" \
  path="/etc/bind/rndc.key" dev="hda" ino=1488 \
  scontext=system_u:system_r:ndc_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:named_zone_t:s0 \
  tclass=file permissive=1

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/contrib/bind.if | 19 +++++++++++++++++++
 policy/modules/system/init.te  |  6 ++++++
 2 files changed, 25 insertions(+)

--- refpolicy.orig/policy/modules/contrib/bind.if
+++ refpolicy/policy/modules/contrib/bind.if
@@ -157,6 +157,25 @@ interface(`bind_read_dnssec_keys',`
 
 ########################################
 ## <summary>
+##	Create dnssec keys in the bind conf
+##      dir with the dnssec keys type.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition.
+##	</summary>
+## </param>
+#
+interface(`bind_dnssec_keys_filetrans',`
+	gen_require(`
+		type named_zone_t, dnssec_t;
+	')
+
+	filetrans_pattern($1, named_zone_t, dnssec_t, file, "rndc.key")
+')
+
+########################################
+## <summary>
 ##	Read bind named configuration files.
 ## </summary>
 ## <param name="domain">
--- refpolicy.orig/policy/modules/system/init.te
+++ refpolicy/policy/modules/system/init.te
@@ -584,6 +584,12 @@ userdom_read_user_home_content_files(ini
 # started from init should be placed in their own domain.
 userdom_use_user_terminals(initrc_t)
 
+optional_policy(`
+	# WRL fix:
+	# create /etc/bind/rndc.key with type dnssec_t
+	bind_dnssec_keys_filetrans(initrc_t)
+')
+
 ifdef(`distro_debian',`
 	kernel_getattr_core_if(initrc_t)
 
